InnerEye/ML/model_training.py

```diff
@@ -89,142 +94,141 @@ def create_lightning_trainer(config: ModelConfigBase,
-        # Save the dataset files for later use in cross validation analysis
-        config.write_dataset_files()
-        logging.info(f"Model checkpoints are saved at {config.checkpoint_folder}")
-
-        # set the random seed for all libraries
-        ml_util.set_random_seed(config.get_effective_random_seed(), "Patch visualization")
-        # Visualize how patches are sampled for segmentation models. This changes the random generator, but we don't
-        # want training to depend on how many patients we visualized, and hence set the random seed again right after.
-        if isinstance(config, SegmentationModelBase):
-            with logging_section("Visualizing the effect of sampling random crops for training"):
-                visualize_random_crops_for_dataset(config)
+    # InnerEye models use this logger for diagnostics
+    if isinstance(lightning_model, InnerEyeLightning):
+        if storing_logger is None:
+            raise ValueError("InnerEye models require the storing_logger for diagnostics")
+        lightning_model.storing_logger = storing_logger
 
-        # Print out a detailed breakdown of layers, memory consumption and time.
-        generate_and_print_model_summary(config, lightning_model.model)
-
-        if config.monitoring_interval_seconds > 0:
-            # initialize and start GPU monitoring
-            gpu_tensorboard = config.logs_folder / "gpu_utilization"
-            # Result file in CSV format should NOT live in the logs folder, the streaming upload that is
-            # used for this folder might corrupt the file.
-            gpu_csv = config.outputs_folder / "gpu_utilization"
-            gpu_csv.mkdir(parents=True, exist_ok=True)
-            logging.info(f"Starting resource monitor. GPU utilization will be written to Tensorboard in "
-                         f"{gpu_tensorboard}, aggregate metrics to {gpu_csv}")
-            resource_monitor = ResourceMonitor(interval_seconds=config.monitoring_interval_seconds,
-                                               tensorboard_folder=gpu_tensorboard,
-                                               csv_results_folder=gpu_csv)
-            resource_monitor.start()
-
-    # Training loop
     logging.info("Starting training")
-
-    lightning_data = TrainingAndValidationDataLightning(config)  # type: ignore
-    # When trying to store the config object in the constructor, it does not appear to get stored at all, later
-    # reference of the object simply fail. Hence, have to set explicitly here.
-    lightning_data.config = config
-    trainer.fit(lightning_model, datamodule=lightning_data)
-    trainer.logger.close()  # type: ignore
-    lightning_model.close_all_loggers()
+    # When training models that are not built-in InnerEye models, we have no guarantee that they write
+    # files to the right folder. Best guess is to change the current working directory to where files should go.
+    with change_working_directory(container.outputs_folder):
```

Comment:
Can this affect any data that the user is trying to read in assuming the repository root as the working directory?

Reply:
No, it should not - the data readers should always consume their data from the folder given in self.local_dataset.